docs: Migrate usage guide

jaideepr97 opened the pull request:
**What type of PR is this?**
> Uncomment only one ` /kind` line, and delete the rest.
> For example, `> /kind bug` would simply become: `/kind bug`

> /kind bug
> /kind cleanup
> /kind failing-test
> /kind enhancement
/kind documentation
> /kind code-refactoring


**What does this PR do / why we need it**:
Migrates the gitops usage guide from internal google doc to markdown

**Have you updated the necessary documentation?**

* [x] Documentation update is required by this PR.
* [x] Documentation has been updated.

**Which issue(s) this PR fixes**:

Fixes https://issues.redhat.com/browse/GITOPS-1692?

**Test acceptance criteria**:

* [ ] Unit Test
* [ ] E2E Test

**How to test changes / Special notes to the reviewer**:
Click on docs > Openshift Gitops Usage Guide.md

iam-veeramalla commented on docs/OpenShift GitOps Usage Guide.md:
```diff
@@ -0,0 +1,1291 @@
+# OpenShift GitOps Usage Guide
+
+Crowd-sourced, work-in-progress, yet-to-be-official
```
I think we can remove this

iam-veeramalla commented:
@jaideepr97 The PR looks good. Thanks for the efforts in migrating the doc in MD file :)

1. Can we setup Integration with Readthedocs ? I am not sure if `redhat-developer` has an account in Readthedocs but the integration is very simple. I see a lot of advantage in this integration.

2. We were also discussing about using GitHub wiki. 
```
You can edit wikis directly on GitHub, or you can edit wiki files locally. By default, only people with write access to your repository can make changes to wikis, although you can allow everyone on GitHub.com to contribute to a wiki in a public repository. For more information, see "[Changing access permissions for wikis](https://docs.github.com/en/communities/documenting-your-project-with-wikis/changing-access-permissions-for-wikis)".
```

jaideepr97 commented:
Thanks @iam-veeramalla, addressed your comment

I agree, having readthedocs integration would be the best long-term option and could be the next step we look into
I believe this story was just about moving the usage guide out of google docs for now 

I think asking users to reference this guide from the release branch is a good intermediate solution while we look into readthedocs integration

reginapizza commented on docs/OpenShift GitOps Usage Guide.md:
````diff
@@ -0,0 +1,1291 @@
+grantMethod: prompt 
+```
+
+1. The name of the OAuth client is used as the client_id parameter when making requests to <namespace_route>/oauth/authorize and <namespace_route>/oauth/token.\
````
remove / at the end

reginapizza commented on docs/OpenShift GitOps Usage Guide.md:
````diff
@@ -0,0 +1,1291 @@
+```
+
+1. The name of the OAuth client is used as the client_id parameter when making requests to <namespace_route>/oauth/authorize and <namespace_route>/oauth/token.\
+2. The secret is used as the client_secret parameter when making requests to <namespace_route>/oauth/token.\
````
remove / at the end

reginapizza commented on docs/OpenShift GitOps Usage Guide.md:
```diff
@@ -0,0 +1,1291 @@
+
+1. The name of the OAuth client is used as the client_id parameter when making requests to <namespace_route>/oauth/authorize and <namespace_route>/oauth/token.\
+2. The secret is used as the client_secret parameter when making requests to <namespace_route>/oauth/token.\
+3. The redirect_uri parameter specified in requests to <namespace_route>/oauth/authorize and <namespace_route>/oauth/token must be equal to or prefixed by one of the URIs listed in the redirectURIs parameter value. Refer to the screenshot in Step 6. \
```
remove / at the end, also what is the screenshot in Step 6? I don't see a step 6 here

reginapizza commented on docs/OpenShift GitOps Usage Guide.md:
```diff
@@ -0,0 +1,1291 @@
+
+![image alt text](assets/image_23.png)
+
+2. Make sure *ArgoCDAdmins *group has required permissions in the argocd-rbac configmap. Please refer to the ArgoCD Rbac documentation [here](https://argoproj.github.io/argo-cd/operator-manual/rbac/).
```
remove space after ArgoCDAdmins, RBAC capitalization

reginapizza commented on docs/OpenShift GitOps Usage Guide.md:
```diff
@@ -0,0 +1,1291 @@
+
+2. Make sure *ArgoCDAdmins *group has required permissions in the argocd-rbac configmap. Please refer to the ArgoCD Rbac documentation [here](https://argoproj.github.io/argo-cd/operator-manual/rbac/).
+
+In this case, I wish to provide admin access to the users in *ArgoCDAdmins *groupargocd-rbac-cmArgoCDAdmins
```
remove space after ArgoCDAdmins, also not sure what the ending of this sentence should be but it definitely got messed up

reginapizza commented on docs/OpenShift GitOps Usage Guide.md:
```diff
@@ -0,0 +1,1291 @@
+
+### Working with Dex
+
+**Note: **For a fresh install of v1.3.0, Dex is automatically configured. You can log into the default Argo CD instance in the openshift-gitops namespace using the OpenShift or kubeadmin credentials. As an admin you can disable the Dex installation after the Operator is installed which will remove the Dex deployment from the openshift-gitops namespace.
```
remove space after Note:

reginapizza commented on docs/OpenShift GitOps Usage Guide.md:
```diff
@@ -0,0 +1,1291 @@
+
+
+
+**For upgrades, **We can enable it by updating the Subscription resource for the OpenShift Gitops Operator.
```
Remove space after `For upgrades, `

reginapizza commented on docs/OpenShift GitOps Usage Guide.md:
```diff
@@ -0,0 +1,1291 @@
+
+`oc patch subscriptions.operators.coreos.com/openshift-gitops-operator -n openshift-operators --type='merge' --patch '{ "spec": { "config": { "env": [ { "name": "DISABLE_DEX", "value": "false" } ] } } }'`
+
+This will get the argocd-cluster-dex-server instance running. To enable login with openshift, you need to update the argo-cd CR with the below field in spec. 
```
OpenShift capitalization

reginapizza commented on docs/OpenShift GitOps Usage Guide.md:
```diff
@@ -0,0 +1,1291 @@
+
+#### Restricting dex / openShiftOAuth to only a set of groups
+
+As discussed here [https://cloud.redhat.com/blog/openshift-authentication-integration-with-argocd](https://cloud.redhat.com/blog/openshift-authentication-integration-with-argocd) you can restrict oauth access to certain groups. Currently it is not possible to restrict the ArgoCD to only a set of groups, through `openShiftOAuth: true`, the RFE is tracked upstream [here](https://github.com/argoproj ...
```
the issue here is closed? So is this point no longer needed?

reginapizza replied:
also looks like something followed it when nothing does now, remove the :

jaideepr97 replied:
I'm not too sure about your first question, leaving it as is for now since thats how its captured in the usage guide

reginapizza replied:
yeah no worries, maybe something that just need to be updated in both.

jaideepr97 replied:
I know that the groups thing works now with dex, but i wasn't sure how that affects the documentation that follows and whether that was originally meant to be a workaround ,and is not needed now or something else

reginapizza commented on docs/OpenShift GitOps Usage Guide.md:
```diff
@@ -0,0 +1,1291 @@
+
+As discussed here [https://cloud.redhat.com/blog/openshift-authentication-integration-with-argocd](https://cloud.redhat.com/blog/openshift-authentication-integration-with-argocd) you can restrict oauth access to certain groups. Currently it is not possible to restrict the ArgoCD to only a set of groups, through `openShiftOAuth: true`, the RFE is tracked upstream [here](https://github.com/argoproj ...
+
+Assuming you have done the above steps to enable openShiftOAuth: true: you can use the following commands to:
```
put in single quotes? `openShiftOAuth: true`

reginapizza commented on docs/OpenShift GitOps Usage Guide.md:
```diff
@@ -0,0 +1,1291 @@
+
+![image alt text](assets/image_25.png)
+
+\
```
remove this \

reginapizza commented on docs/OpenShift GitOps Usage Guide.md:
```diff
@@ -0,0 +1,1291 @@
+
+[https://github.com/redhat-developer/kam/tree/master/docs/journey/day1](https://github.com/redhat-developer/kam/tree/master/docs/journey/day1)
+
+*( contents of the link should be expanded here ) *
```
remove this

reginapizza commented on docs/OpenShift GitOps Usage Guide.md:
```diff
@@ -0,0 +1,1291 @@
+
+*( contents of the link should be expanded here ) *
+
+* * *
```
remove

reginapizza commented on docs/OpenShift GitOps Usage Guide.md:
```diff
@@ -0,0 +1,1291 @@
+
+## Setting up a new ArgoCD instance
+
+GitOps Operator installs an instance of ArgoCD in **openshift-gitops **namespace with additional permissions, which allows it to manage certain cluster scoped resources.
```
remove space after openshift-gitops

reginapizza commented on docs/OpenShift GitOps Usage Guide.md:
```diff
@@ -0,0 +1,1291 @@
+
+#### ArgoCD Installation
+
+        * To install an ArgoCD instance, go to Installed Operators.
+
+        * Create a new project**_ _**or select an existing project where you want to install the ArgoCD instance.
```
remove indents and probably replace * with numbers?

reginapizza replied:
remove **_ _** after project

reginapizza commented on docs/OpenShift GitOps Usage Guide.md:
```diff
@@ -0,0 +1,1291 @@
+
+* Click on "Create" button to create ArgoCD Instance and specify following configuration:
+
+    * Name: <argocd-instance-name>
```
content within tags don't show up

reginapizza commented on docs/OpenShift GitOps Usage Guide.md:
```diff
@@ -0,0 +1,1291 @@
+
+To grant ArgoCD the permissions to manage resources in multiple namespaces, we need to configure the namespace with a label **"argocd.argoproj.io/managed-by"** and the value being the **namespace** of the ArgoCD instance meant to manage the namespace. 
+
+**Example**: For ArgoCD instance deployed in the namespace `**foo**` wants to manage resources in namespace `**bar**`.
```
remove *'s around foo and bar

reginapizza commented on docs/OpenShift GitOps Usage Guide.md:
```diff
@@ -0,0 +1,1291 @@
+
+`oc apply -f namespace.yaml`
+
+Via Openshift Console - 
```
Make this a header probably?

reginapizza commented on docs/OpenShift GitOps Usage Guide.md:
```diff
@@ -0,0 +1,1291 @@
+
+Via Openshift Console - 
+
+On Openshift Console, as an admin, navigate to - 
+
+**User Management -> Role Bindings ->** select **Create Binding.**
```
make this same line, remove -

reginapizza commented on docs/OpenShift GitOps Usage Guide.md:
```diff
@@ -0,0 +1,1291 @@
+
+    * If the Project is already selected as the one you want to create RoleBinding for, it does not ask for Binding Type and assumes it to be a RoleBinding. 
+
+* Under Role Binding, provide a RoleBinding **Name** and **Namespace **(the one you are granting permissions for).
```
remove space after Namespace

reginapizza commented on docs/OpenShift GitOps Usage Guide.md:
```diff
@@ -0,0 +1,1291 @@
+
+#### Deploy resources to a different namespace with custom role
+
+As an administrative user, when you give Argo CD access to a namespace by using the argocd.argoproj.io/managed-by label, it assumes namespace-admin privileges. These privileges are an issue for administrators who provide namespaces to non-administrators, such as development teams, because the privileges enable non-administrators to modify objects such as network policies.
```
single quotes around `rgocd.argoproj.io/managed-by`?

reginapizza commented on docs/OpenShift GitOps Usage Guide.md:
```diff
@@ -0,0 +1,1291 @@
+
+As an administrative user, when you give Argo CD access to a namespace by using the argocd.argoproj.io/managed-by label, it assumes namespace-admin privileges. These privileges are an issue for administrators who provide namespaces to non-administrators, such as development teams, because the privileges enable non-administrators to modify objects such as network policies.
+
+With this update, administrators can configure a common cluster role for all the managed namespaces. In role bindings for the Argo CD application controller, the Operator refers to the CONTROLLER_CLUSTER_ROLE environment variable. In role bindings for the Argo CD server, the Operator refers to the SERVER_CLUSTER_ROLE environment variable. If these environment variables contain custom roles, the O ...
```
single quotes around `CONTROLLER_CLUSTER_ROLE` and `SERVER_CLUSTER_ROLE`?

reginapizza commented on docs/OpenShift GitOps Usage Guide.md:
```diff
@@ -0,0 +1,1291 @@
+
+### Cluster Configuration
+
+When the user wishes to install ArgoCD with the purpose of managing OpenShift cluster resources, ArgoCD is granted permissions to [manage specific cluster-scoped resources](https://docs.google.com/document/d/1HncLIPlUkO5rfTHCi4zAygB6Xx6K3dAgI5PM5GT1G6A/edit#heading=h.e3snv4wit2fc) which include [platform operators](https://docs.openshift.com/container-platform/4.6/architecture/control-plane.html# ...
```
Argo CD not ArgoCD

reginapizza commented on docs/OpenShift GitOps Usage Guide.md:
```diff
@@ -0,0 +1,1291 @@
+Argo CD is granted the following permissions to manage specific cluster-scoped resources which include platform operators, optional OLM operators and user management. **Argo CD is not granted cluster-admin**.
+
+
+User can extend the permissions provided to ArgoCD instance by following the steps provided in this [section](#heading=h.8m0tejtgffa7) - 
```
remove -

reginapizza commented on docs/OpenShift GitOps Usage Guide.md:
```diff
@@ -0,0 +1,1291 @@
+
+#### Argo CD Installation
+
+To manage cluster-config, deploy an ArgoCD instance using the steps provided above. 
+
+    * As an admin, update the existing Subscription Object for Gitops Operator and add ARGOCD_CLUSTER_CONFIG_NAMESPACES to the spec.
+
+   On Openshift Console, go to 
+
+    * **Administration -> CustomResourceDefinition -> Subscription -> Instances **and select **"****openshift-gitops-operator****".**
```
change to 
```
To manage cluster-config, deploy an ArgoCD instance using the steps provided above. 

* As an admin, update the existing Subscription Object for Gitops Operator and add `ARGOCD_CLUSTER_CONFIG_NAMESPACES` to the spec.

* On Openshift Console, go to * **Administration -> CustomResourceDefinition -> Subscription -> Instances** and select **"openshift-gitops-operator".**
```

reginapizza commented on docs/OpenShift GitOps Usage Guide.md:
```diff
@@ -0,0 +1,1291 @@
+
+    * **Administration -> CustomResourceDefinition -> Subscription -> Instances **and select **"****openshift-gitops-operator****".**
+
+**
```
remove this *'s

reginapizza commented on docs/OpenShift GitOps Usage Guide.md:
```diff
@@ -0,0 +1,1291 @@
+
+* Click on **Create**
+
+**	**Via CLI - 
```
Remove *'s, make into a header, same as "Via OpenShift Console" above

reginapizza commented on docs/OpenShift GitOps Usage Guide.md:
```diff
@@ -0,0 +1,1291 @@
+
+##### 	Create Cluster Role Binding
+
+	Via CLI - 
```
Make into a header, remove indent

reginapizza commented on docs/OpenShift GitOps Usage Guide.md:
```diff
@@ -0,0 +1,1291 @@
+`oc create -f cluster_role_binding.yaml`
+	
+
+Via OpenShift Console -
```
make into a header

reginapizza commented on docs/OpenShift GitOps Usage Guide.md:
```diff
@@ -0,0 +1,1291 @@
+
+`kubectl -n argocd patch argocd example --type='json' -p='[{"op": "replace", "path": "/spec/controller/resources/limits/memory", "value":"2048Mi"}]'`
+
+**The above commands can be modified to replace application controller with any other OpenShift GitOps workloads such as ApplicationSet Controller, Dex, Redis, Repo Server, Server, Keycloak e.t.c.,. **
```
remove space at the end

reginapizza commented on docs/OpenShift GitOps Usage Guide.md:
```diff
@@ -0,0 +1,1291 @@
+
+![image alt text](assets/image_32.png)
+
+To take full advantage of the metrics provided by ArgCD, you can install Grafana to help analyze and visualize the metrics.  This [blog post](https://www.redhat.com/en/blog/custom-grafana-dashboards-red-hat-openshift-container-platform-4) walks you through deploying a custom Grafana instance to your cluster using the community provided Grafana operator.  *Note that community Operators are operato ...
```
"Argo CD"

jaideepr97 commented:
/retest v4.7-e2e

jaideepr97 commented:
/test v4.7-e2e

reginapizza commented on docs/OpenShift GitOps Usage Guide.md:
```diff
@@ -0,0 +1,1291 @@
+
+![image alt text](assets/image_33.png)
+
+* If this is the first time the Kibana dashboard is launched, an index pattern needs to be created in Kibana.  A simple "***" **index will do for now.![image alt text](assets/image_34.png)
```
change to 
```
A simple `"*"` index will do for now.
```

reginapizza commented on docs/OpenShift GitOps Usage Guide.md:
```diff
@@ -0,0 +1,1291 @@
+
+* Under the search bar, click on "add a filter".
+
+* Provide ‘kubernetes.namespace_name’ as the filter with value ‘openshift-gitops’. This filter would retrieve logs for all the pods in "openshift-gitops" namespace. ![image alt text](assets/image_36.png)  
```
the single quotes are not getting applied right here

reginapizza commented on docs/OpenShift GitOps Usage Guide.md:
```diff
@@ -0,0 +1,1291 @@
+
+* Provide ‘kubernetes.namespace_name’ as the filter with value ‘openshift-gitops’. This filter would retrieve logs for all the pods in "openshift-gitops" namespace. ![image alt text](assets/image_36.png)  
+
+* To retrieve logs for particular pods, additional filters like **_kubernetes.pod_name _**can be added to the filter list.
```
change to 
```
* To retrieve logs for particular pods, additional filters like **kubernetes.pod_name** can be added to the filter list.
```

reginapizza commented on docs/OpenShift GitOps Usage Guide.md:
```diff
@@ -0,0 +1,1291 @@
+
+## Health status of OpenShift resources
+
+This enhancement adds support to display the Health status of OpenShift resources like DeploymentConfig, routes and Operators that you install using OLM.. This enables you to better monitor the overall health status of your application.
```
extra period here after OLM

reginapizza commented on docs/OpenShift GitOps Usage Guide.md:
```diff
@@ -0,0 +1,1291 @@
+
+This is a breaking change and needs some manual steps to be performed before the upgrade. 
+
+1. Before the upgrade, store the content (data) of **argocd-cm **locally. 
```
remove space after argocd-cm

reginapizza commented on docs/OpenShift GitOps Usage Guide.md:
```diff
@@ -0,0 +1,1291 @@
+
+1. Before the upgrade, store the content (data) of **argocd-cm **locally. 
+
+2. Delete the **argocd-cluster **instance (default Argo CD instance) that is present in the cluster. 
```
remove space after argocd-cluster

reginapizza commented on docs/OpenShift GitOps Usage Guide.md:
```diff
@@ -0,0 +1,1291 @@
+
+3. Upgrade the GitOps Operator. 
+
+4. Apply the patch manually from the **argocd-cm **for the resources that were previously present on the previous Argo CD instance
```
remove space after argocd-cm

reginapizza reviewed (changes requested):
Mostly nits with markdown formatting! Sorry there's so many >.<

jaideepr97 commented on docs/OpenShift GitOps Usage Guide.md:
```diff
@@ -0,0 +1,1291 @@
+
+   On Openshift Console, go to 
+
+    * **Administration -> CustomResourceDefinition -> Subscription -> Instances **and select **"****openshift-gitops-operator****".**
```
bless your soul for sharing the correct markdown as well :pray:

jaideepr97 commented:
@wtam2018 could you PTAL

iam-veeramalla commented:
/approve
/lgtm
